File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Java_Solutions/Problem2.java
public class Problem2 {
    public static void main(String[] args) {
        double x0 = 2.0;  // Initial guess
        double tolerance = 0.0001;
        int maxIterations = 10;
        
        System.out.println("Newton-Raphson Method for f(x) = x³ - 2x - 5 = 0");
        System.out.println("Initial guess: x₀ = " + x0);
        System.out.println("\nIterations:");
        
        double x = x0;
        for (int i = 0; i < maxIterations; i++) {
            double fx = x * x * x - 2 * x - 5;  // f(x) = x³ - 2x - 5
            double fpx = 3 * x * x - 2;         // f'(x) = 3x² - 2
            
            double xNext = x - fx / fpx;
            
            System.out.printf("Iteration %d: x = %.6f, f(x) = %.6f\n", 
                            i + 1, x, fx);
            
            if (Math.abs(xNext - x) < tolerance) {
                System.out.printf("\nRoot found: %.6f\n", xNext);
                break;
            }
            
            x = xNext;
        }
    }
} 

File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Python_Solutions/Problem2.py
def newton_raphson(x0, tolerance=0.0001, max_iterations=10):
    x = x0
    print(f"Newton-Raphson Method for f(x) = x³ - 2x - 5 = 0")
    print(f"Initial guess: x₀ = {x0}")
    print("\nIterations:")
    
    for i in range(max_iterations):
        fx = x**3 - 2*x - 5  # f(x) = x³ - 2x - 5
        fpx = 3*x**2 - 2     # f'(x) = 3x² - 2
        
        x_next = x - fx / fpx
        
        print(f"Iteration {i+1}: x = {x:.6f}, f(x) = {fx:.6f}")
        
        if abs(x_next - x) < tolerance:
            print(f"\nRoot found: {x_next:.6f}")
            break
            
        x = x_next

if __name__ == "__main__":
    newton_raphson(2.0)  # Initial guess 